### DISCUSSION TITLE
Support for DateOnly and TimeOnly

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#686] [General]
I have a use case for DateOnly but I'm running into issue when syncing, I think its down to the de\serialization.

I can see that Newtonsoft JSON supports these from 13.0.2 and .NET from version 7

What does this lib use ? are there plans .NET 7

**Comment by adrianhall** [comment]
The server side should be supported no problem (DateOnly is supported from .NET 6 actually).

The client side requires us to support .NET Standard 2.0 because of Xamarin support, which does not include DateOnly.  I'm looking at the appropriate point at which to drop Xamarin support and move the library to min .NET 6.  At that point, I'll be able to support DateOnly and TimeOnly.

There are no plans to support STS releases of .NET - I'm going to update the client and server to support .NET 8 when it reaches GA in November.

**Comment by henda79** [comment]
I've just used DateTime as the client hangs when calling `GetAsyncItems` when the Sqlite table is created from a model using DateOnly, I'm not sure what type is used internally for the column but the are no exceptions when inserting or updating, it just hangs when getting rows from the offline table.

Personally I need the .NET standard 2.0 as I'm part way through a project using Xamarin.Forms and don't really want to migrate to MAUI as I not had a good experience with it so far, it seems to still have a lot of niggles and workarounds, it's like the early Xamarin days all over again.

**Reply by adrianhall** [reply]
Yeah- the client doesn't know how to deal with DateOnly.  It shouldn't hang - it should either throw or complete.  If it is hanging, file a bug with a repro.

**Comment by henda79** [comment]
Will do once I get through my backlog.
